src/bun.zig

```diff
@@ -2091,8 +2091,46 @@ pub const Stat = if (Environment.isWindows) windows.libuv.uv_stat_t else std.os.
 
 pub var argv: [][:0]const u8 = &[_][:0]const u8{};
 
+extern "c" fn CommandLineToArgvW([*:0]u16, *c_int) ?[*][*:0]u16;
+
 pub fn initArgv(allocator: std.mem.Allocator) !void {
-    argv = try std.process.argsAlloc(allocator);
+    if (comptime !Environment.isWindows) {
+        argv = try std.process.argsAlloc(allocator);
+    } else {
+        // Zig's implementation of `std.process.argsAlloc()`on Windows platforms
+        // is not reliable, specifically the way it splits the command line string.
+        //
+        // For example, an arg like "foo\nbar" will be
+        // erroneously split into two arguments: "foo" and "bar".
+        //
+        // To work around this, we can simply call the Windows API functions
+        // that do this for us.
+        //
+        // Updates in Zig v0.12 related to Windows cmd line parsing may fix this,
+        // see (here: https://ziglang.org/download/0.12.0/release-notes.html#Windows-Command-Line-Argument-Parsing),
+        // so this may only need to be a temporary workaround.
+        const cmdline_ptr = std.os.windows.kernel32.GetCommandLineW();
+        var length: c_int = 0;
+        const argvu16_ptr = CommandLineToArgvW(cmdline_ptr, &length) orelse {
```

Comment:
outofmemory is handled here, which implies allocation, but i dont see this being freed. i dont think its a big issue since its a small allocation with a max upper bound of 64k.

Reply:
According to the GetCommandLineW [docs](https://learn.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getcommandlinew#remarks) there's no need to free it:
> The lifetime of the returned value is managed by the system, applications should not free or modify this value.

Added a comment noting that

Reply:
it should never be freed